cmake/SpicyAnalyzer.cmake

```diff
@@ -0,0 +1,26 @@
+function(spicy_analyzer)
+    set(options)
+    set(oneValueArgs NAME KIND)
+    set(multiValueArgs SOURCES SCRIPTS)
+
+    cmake_parse_arguments(SPICY_ANALYZER "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if (NOT DEFINED SPICY_ANALYZER_NAME)
+        message(FATAL_ERROR "NAME is required")
+    endif()
+
+    if (NOT DEFINED SPICY_ANALYZER_KIND)
+        message(FATAL_ERROR "KIND is required")
+    endif()
+
+    spicy_add_analyzer("${SPICY_ANALYZER_NAME}" "${SPICY_ANALYZER_SOURCES}")
```

Comment:
To avoid confusion, we should probably deprecate `spicy_add_analyzer` and eventually either remove it or make it internal.

Reply:
When adding this function to zeek/spicy-plugin in zeek/spicy-plugin#61 I added a mode where one can use either the new form with named args, or the previous form with unnamed args. This should help avoid any confusion.